packages/headless/src/api/search/search/result.ts

```diff
@@ -172,4 +172,14 @@ export interface Result {
    * TopResult
    */
   rankingModifier?: string;
+
+  /**
+   * The child results as returned in a product grouping context.
+   */
+  childResults: Result[];
```

Comment:
I don't think we should add `childResults` and `totalNumberOfChildResults` to generic results.

For generic results, we expect developers to use `buildFoldedResultList` and the `folding` slice, since we try to resolve the initial hierarchy of indexed results.

Reply:
Agreed, I'd prefer to not change this interface too,